java-frontend/src/main/java/org/sonar/java/JavaFrontend.java

```diff
@@ -0,0 +1,643 @@
+          break;
+        case ARRAY_ACCESS_EXPRESSION:
+          executeArrayAccessExpression((ArrayAccessExpressionTree) tree);
+          break;
+        case NEW_ARRAY:
+          executeNewArray((NewArrayTree) tree);
+          break;
+        case NEW_CLASS:
+          executeNewClass((NewClassTree) tree);
+          break;
+        case MULTIPLY:
+        case DIVIDE:
+        case REMAINDER:
+        case PLUS:
+        case MINUS:
+        case LEFT_SHIFT:
+        case RIGHT_SHIFT:
+        case UNSIGNED_RIGHT_SHIFT:
+        case AND:
+        case XOR:
+        case OR:
+        case GREATER_THAN:
+        case GREATER_THAN_OR_EQUAL_TO:
+        case LESS_THAN:
+        case LESS_THAN_OR_EQUAL_TO:
+        case EQUAL_TO:
+        case NOT_EQUAL_TO:
+          executeBinaryExpression(tree);
+          break;
+        case POSTFIX_INCREMENT:
+        case POSTFIX_DECREMENT:
+        case PREFIX_INCREMENT:
+        case PREFIX_DECREMENT:
+        case UNARY_MINUS:
+        case UNARY_PLUS:
+        case BITWISE_COMPLEMENT:
+        case LOGICAL_COMPLEMENT:
+        case INSTANCE_OF:
+          executeUnaryExpression(tree);
+          break;
+        case IDENTIFIER:
+          executeIdentifier((IdentifierTree) tree);
+          break;
+        case MEMBER_SELECT:
+          executeMemberSelect((MemberSelectExpressionTree) tree);
+          break;
+        case INT_LITERAL:
+        case LONG_LITERAL:
+        case FLOAT_LITERAL:
+        case DOUBLE_LITERAL:
+        case CHAR_LITERAL:
+        case STRING_LITERAL:
+        case BOOLEAN_LITERAL:
+        case NULL_LITERAL:
+          executeLiteral();
+          break;
+        case LAMBDA_EXPRESSION:
+        case METHOD_REFERENCE:
+          executeMethodReference();
+          break;
+        case ASSERT_STATEMENT:
+          executeAssertStatement();
+          break;
+        default:
+          throw new IllegalArgumentException("Unhandled tree: " + tree.getClass().getSimpleName() + " - " + tree);
+      }
+    }
+
+    private void executeLiteral() {
+      stack.push(new TaintFreeSource());
+    }
+
+    private void executeIdentifier(IdentifierTree tree) {
+      if (TaintProgramState.isString(tree.symbolType())) {
+        stack.push(state.taintSourceFor(tree.symbol()));
+      } else {
+        stack.push(new TaintFreeSource());
+      }
+    }
+
+    private void executeAssignment(AssignmentExpressionTree tree) {
+      TaintSource ts;
+      if (tree.is(Tree.Kind.ASSIGNMENT)) {
+        ts = stack.pop();
+        if (!ExpressionUtils.isSimpleAssignment(tree)) {
+          stack.pop();
+        }
+      } else {
+        ts = stack.pop();
+        stack.pop();
+      }
+
+      Symbol symbol = null;
+      if (tree.variable().is(Tree.Kind.IDENTIFIER) || ExpressionUtils.isSelectOnThisOrSuper(tree)) {
+        symbol = ExpressionUtils.extractIdentifier(tree).symbol();
+        // TODO How to handle any assignment?
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135)